### ISSUE TITLE
Document performance increase from not using NanoDate()

### ISSUE THREAD

**Issue by dandv** [node-influx/node-influx#216]
The [Usage](https://node-influx.github.io/manual/usage.html) page states that,

> The exception is if you manually ask for a precision of `ms` or coarser in the options parameter of Influx#query method. Then, we parse your times into normal Dates.

For certain classes of applications, nanosecond-precision is overkill. Since performance is a focus in v5, it would be interesting to document the potential gains coming from sticking with `Date`s.

**Comment by connor4312** [comment]
Sure. I'm not sure there's too many gains there, I made sure to keep the overhead as minimal as possible -- right now it just [attaches three properties](https://github.com/node-influx/node-influx/blob/master/src/grammar/times.ts#L290) to the Date instance, we don't do any fancy parsing on it until we need to.

I've been thinking it might make sense to always parse to a NanoDate regardless of the requested precision. This would make it easier for consumers to use without all their business logic having to be acutely aware of how Influx is queried or risk runtime errors. If benchmarks show, as I think they will, that there's little to no overhead, I'll make that modification.

**Comment by connor4312** [comment]
I tested the cases where plain Dates could be used instead of NanoDates. 

```
       7,983,418 op/s » parsing millisecond timestamp to plain date
       4,580,856 op/s » parsing millisecond timestamp to nanodate
       7,190,202 op/s » parsing hours timestamp to plain date
       4,620,127 op/s » parsing hours timestamp to nanodate
```

Parsing to standard dates is about 75% faster. This has a nearly proportional impact on the result parsing suite:

```
           1,938 op/s » parsing (ms) [ 3,876,000 rows/sec ]
           1,414 op/s » parsing (ns) [ 2,828,000 rows/sec ]
```

(Interestingly this indicates that the overwhelming majority of time is spent in date parsing, meaning that parsing is at or near the maximum speed it could possibly be (except if parsing were to be dropped).)

I think having Dates be parsed as their own thing is probably fine, but as usual I'm totally open to feedback on this. My own use cases of node-influx are an infinitely small minority of the possible use cases 😄

**Comment by connor4312** [comment]
I've managed to squeeze it down to a 14% difference. This is acceptable, I think. If people can't afford that hit they probably should be using queryRaw instead. Still parses well over 3 million rows/sec on my machine.

```
          3,813 op/s » parsing ms
           3,337 op/s » parsing ns
```

**Comment by connor4312** [comment]
Done in https://github.com/node-influx/node-influx/commit/d1cb4ac132def37ce52e966dded2439187c263e0

**Comment by connor4312** [comment]
Published in `5.0.0-alpha.2`, available under `npm install --save influx@next`
